### PR TITLE
Themes: sheet SEO

### PR DESCRIPTION
**Pull request by budzanowski**
This PR is created to address issue #3658.

How to test:
1. Open theme sheet, for example /theme/twentysixteen
2. Open _source_
3. Look for og:title, og:descripton, og:url

Those 3 meta should all have meaningful(not generic) content for that theme.

Test live: https://calypso.live/?branch=fix/3658-sheet-seo

### CODE REVIEW

**Comment by ockham** [comment]
To get this to work, you'll need to change [this line](https://github.com/Automattic/wp-calypso/blob/700ca07adb57b56680b91a72b2b88306ce94a463/client/my-sites/themes/head.jsx#L16) to be analog to the one above.

**Comment by folletto** [comment]
I just noticed the other issue doesn't mention `og:image`. Let's make sure that we generate a great preview — check with the [Fb URL linter](https://developers.facebook.com/tools/debug/og/object/).

**Comment by budzanowski** [comment]
@folletto What about og:site_name?

**Comment by folletto** [comment]
Good points:

* `og:site_name` seems appropriate, let's go with "WordPress.com"
* `og:type` is a bit harder... "article" seems a good match, however it also [states](http://ogp.me/#types): _"Any non-marked up webpage should be treated as og:type website"_, so I'd probably err in starting with "website".

File: client/my-sites/theme/controller.jsx [hunk 1]
```diff
@@ -77,7 +77,10 @@ export function fetchThemeDetailsData( context, next ) {
 export function details( context, next ) {
 	const { slug } = context.params;
 	const user = getCurrentUser( context.store.getState() );
-	const themeName = ( getThemeDetails( context.store.getState(), slug ) || false ).name;
+	const themeDetails = getThemeDetails( context.store.getState(), slug ) || false;
+	const themeName = themeDetails.name;
+	const themeDescription = themeDetails.description;
+	const themeCanonicalUrl = `https://wordpress.com/theme/${ slug }`;
```
Comment (ockham) [inline client/my-sites/theme/controller.jsx]: In the long run, I'd love us to use `getDetailsUrl()` here (which isn't possible now as it expects an entire theme object, not just a slug). Could you add a `// TODO: ...` comment to that effect find this instance easier once I've tweaked `getDetailsUrl()`?

Also, I think it's fine to directly set the `props`' object `description` and `canonicalUrl` fields, without the intermediary constants.

Comment (budzanowski) [inline client/my-sites/theme/controller.jsx]: Done & done.

File: client/my-sites/themes/head.jsx [hunk 2]
```diff
@@ -10,14 +10,17 @@ import React from 'react';
...
-const ThemesHead = ( { title, tier, children } ) => (
+const ThemesHead = ( { title, description, canonicalUrl, type, image, tier, children } ) => (
 	<Head
 		title={ title ? title : get( 'title', tier ) }
-		description={ get( 'description', tier ) }
-		canonicalUrl={ get( 'canonicalUrl', tier ) } >
+		description={ description ? description : get( 'description', tier ) }
+		canonicalUrl={ canonicalUrl ? canonicalUrl : get( 'canonicalUrl', tier ) }
+		type={ type ? type : 'website' }
+		site_name={ 'WprdPress.com' }
```
Comment (ockham) [inline client/my-sites/themes/head.jsx]: Typo :-)

Comment (budzanowski) [inline client/my-sites/themes/head.jsx]: Well spotted.

**Comment by seear** [comment]
Looks good to merge 👍

**Pull request merged**